src/google/protobuf/compiler/cpp/cpp_file.cc

```diff
@@ -599,7 +599,7 @@ void FileGenerator::GenerateBuildDescriptors(io::Printer* printer) {
     // string literal.
     if (breakdown_large_file && file_data.size() > 65535) {
       printer->Print(
-          "static const char descriptor[] = {\n");
+          "static const signed char descriptor[] = {\n");
```

Comment:
The generated code in question doesn't actually use a string literal; it's assigning individual bytes.

I got a bit confused in my commit message; the issue is with unsigned chars and negative values. `const char value[] = {-1}` does not work in C++11. http://stackoverflow.com/questions/4434140/narrowing-conversions-in-c0x-is-it-just-me-or-does-this-sound-like-a-breakin has details on why.

Simple example (Debian G++ 4:4.9.2-2 amd64):
```cpp
int main() {
  const char test[] = {-4};
}
```
```console
$ g++-4.9 -std=c++11 -fno-signed-char test_char.cc
test_char.cc: In function ‘int main()’:
test_char.cc:2:26: warning: narrowing conversion of ‘-4’ from ‘int’ to ‘const char’ inside { } [-Wnarrowing]
   const char test[] = {-4};
                          ^
```

On some architectures, unsigned characters are the default so you don't have to pass the flag to see this issue.